airbyte-commons-protocol/src/main/java/io/airbyte/commons/protocol/AirbyteMessageVersionedMigrator.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.protocol;
+
+import io.airbyte.commons.version.Version;
+import io.airbyte.protocol.models.v0.AirbyteMessage;
+import lombok.AllArgsConstructor;
+
+/**
+ * Wraps message migration from a fixed version to the most recent version
+ */
+@AllArgsConstructor
```

Comment:
Nit: If we imagine that this class may become a singleton at some point, I would recommend a manually declared all argument constructor instead of using the Lombok annotation.

Reply:
done